test/testutils.jl

```diff
@@ -146,14 +161,30 @@ function test_samples(s::Sampleable{Univariate, Discrete},      # the sampleable
     # generate samples using RNG passed or default RNG
     # we also check reproducibility
     if rng === missing
+        Random.seed!(1234)    
+        samples = if !call_scalar
+            rand(s, n)
+        else
+            map((_) -> rand(s), 1:n)
+        end
         Random.seed!(1234)
-        samples = rand(s, n)
-        Random.seed!(1234)
-        samples2 = rand(s, n)
+        samples2 = if !call_scalar
+            rand(s, n)
+        else
+            map((_) -> rand(s), 1:n)
+        end
```

Comment:
Can we introduce `samples3` and `samples4` unconditionally, that sample with `rand`? And then check their equality (`samples3 == samples4`) and the properties below as well?